composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextFields.kt

```diff
@@ -0,0 +1,194 @@
+
+@Preview(showBackground = true)
+@Composable
+fun ErrorPreview() {
+    AppTheme {
+        Column {
+            TextFields.EditText(
+                input = "Test",
+                topLabel = "First Name",
+                onInputChanged = { _ -> },
+                errorMessage = "First name is required"
+            )
+
+            TextFields.EmailTextField(email = "test@test.com", isValid = true, onInputChanged = {})
+            TextFields.PasswordTextField(
+                currentPassword = "test@test.com",
+                isValid = true,
+                onInputChanged = {},
+                onTogglePasswordVisibility = { },
+                isPasswordVisible = false
+            )
+        }
+    }
+}
+
+object TextFields {
+
+    @Composable
+    fun EditText(
+        modifier: Modifier = Modifier,
+        input: String,
+        onInputChanged: (String) -> Unit,
+        topLabel: String? = null,
+        enabled: Boolean = true,
+        errorMessage: String? = null,
+        colors: TextFieldColors = OutlinedTextFieldDefaults.colors(
+            focusedBorderColor = Color.Transparent,
+            unfocusedBorderColor = Color.White,
+            errorContainerColor = Color.White,
+            errorLeadingIconColor = Color.Black,
+            unfocusedContainerColor = Color.White,
+            focusedContainerColor = Color.White,
+            errorBorderColor = Color.Red
+        ),
+        textBoxHeight: Dp = OutlinedTextFieldDefaults.MinHeight,
+        visualTransformation: VisualTransformation = VisualTransformation.None,
+        trailingIcon: @Composable (() -> Unit)? = null,
+        leadingIcon: @Composable (() -> Unit)? = null,
+        hint: String = stringResource(R.string.type_here),
+        maxLines: Int = Int.MAX_VALUE,
+    ) {
+
+        Column(modifier) {
+            topLabel?.let {
+                Text(
+                    text = it,
+                    modifier = Modifier.padding(bottom = 5.dp),
+                    style = TextStyles.topLabelText
+                )
+            }
+
+            OutlinedTextField(
+                modifier = Modifier
+                    .height(textBoxHeight)
+                    .fillMaxWidth(),
+                isError = errorMessage != null,
+                placeholder = {
+                    Text(
+                        text = hint,
+                        style = TextStyles.hint
+                    )
+                },
+                maxLines = maxLines,
+                visualTransformation = visualTransformation,
+                enabled = enabled,
+                shape = RoundedCornerShape(10.dp),
+                colors = colors,
+                trailingIcon = trailingIcon,
+                leadingIcon = leadingIcon,
+                value = input,
+                onValueChange = onInputChanged,
+            )
+
+            errorMessage?.let {
+                Spacer(modifier = Modifier.padding(top = 5.dp))
```

Comment:
I think this padding could be added to the row itself so you don't need an extra spacer, correct me if I'm wrong